base/compiler/abstractinterpretation.jl

```diff
@@ -1033,9 +1033,35 @@ function abstract_call(interp::AbstractInterpreter, fargs::Union{Nothing,Vector{
             add_remark!(interp, sv, "Could not identify method table for call")
             return CallMeta(Any, false)
         end
-        return abstract_call_gf_by_type(interp, nothing, argtypes, argtypes_to_type(argtypes), sv, max_methods)
+        callinfo = abstract_call_gf_by_type(interp, nothing, argtypes, argtypes_to_type(argtypes), sv, max_methods)
+        return callinfo_from_interprocedural(callinfo, fargs)
+    end
+    callinfo = abstract_call_known(interp, f, fargs, argtypes, sv, max_methods)
+    return callinfo_from_interprocedural(callinfo, fargs)
+end
+
+function callinfo_from_interprocedural(callinfo::CallMeta, ea::Union{Nothing,Vector{Any}})
+    if ea === nothing
+        return callinfo
+    end
+
+    rt = callinfo.rt
+    if isa(rt, InterConditional)
+        # try to convert interprocedural conditional constraint from callee into the constraint
+        # on slots of the current frame
+        i = rt.slot
+        if checkbounds(Bool, ea, i)
```

Comment:
I believe we don't need this bound check since `InterConditional` only comes from a "valid" call and its slot is supposed to be always within the bound of the call argument (I confirmed tests got passed w/o this).
I added this just for robustness and safety, but I'm not sure whether or not we should remove this. What do you think ?